upgrade-tidb-using-tiup.md

```diff
@@ -1,192 +1,185 @@
+This document is targeted for users who want to upgrade from TiDB 4.0 versions to TiDB 5.0 versions, or from TiDB 5.0 to a later version.
 
-If you have deployed the TiDB cluster using TiDB Ansible, you can use TiUP to import the TiDB Ansible configuration and perform the upgrade.
+If your cluster to upgrade is v3.1 or an earlier version (v3.0 or v2.1), you need to upgrade your cluster first to v4.0 and then to v5.0. You cannot upgrade your cluster across a major version or versions.
 
 ## Upgrade caveat
 
-- After the upgrade, rolling back to 3.0 or earlier versions is not supported.
-- To update versions earlier than 3.0 to 4.0, first update this version to 3.0 using TiDB Ansible, and use TiUP to import the TiDB Ansible configuration and update the 3.0 version to 4.0.
-- After the TiDB Ansible configuration is imported into and managed by TiUP, you can no longer operate on the cluster using TiDB Ansible. Otherwise, conflicts might occur because of the inconsistent metadata.
-- Currently, you cannot import the TiDB Ansible configuration if the cluster deployed using TiDB Ansible meets one of the following situations:
-    - The TLS encryption is enabled for the cluster.
-    - This is a pure key-value cluster (cluster with no TiDB instance).
-    - `Kafka` is enabled for the cluster.
-    - `Spark` is enabled for the cluster.
-    - `Lightning` / `Importer` is enabled for the cluster.
-    - You still use the `'push'` method to collect monitoring metrics (since v3.0, `pull` is the default mode, which is supported if you have not modified this mode).
-    - In the `inventory.ini` configuration file, the `node_exporter` or `blackbox_exporter` item of the machine is set to non-default ports through `node_exporter_port` or `blackbox_exporter_port`, which is compatible if you have unified the configuration in the `group_vars` directory. Or the `node_exporter` or `blackbox_exporter` item of a machine is set to a `deploy_dir` that is different from that of other machines.
+- After the upgrade, rolling back to 4.0 or earlier versions is not supported.
+- For the v4.0 cluster managed using TiDB Ansible, you need to import the cluster to TiUP (`tiup cluster`) for new management according to [Upgrade TiDB Using TiUP (v4.0)](https://docs.pingcap.com/tidb/v4.0/upgrade-tidb-using-tiup#import-tidb-ansible-and-the-inventoryini-configuration-to-tiup). Then you can upgrade the cluster to v5.0 according to this document.
+- To update versions earlier than 3.0 to 5.0, first update this version to 3.0 using [TiDB Ansible](https://docs.pingcap.com/tidb/v3.0/upgrade-tidb-using-ansible), use TiUP (`tiup cluster`) to import the TiDB Ansible configuration and, update the 3.0 version to 4.0 according to [Upgrade TiDB Using TiUP (v4.0)](https://docs.pingcap.com/tidb/v4.0/upgrade-tidb-using-tiup#import-tidb-ansible-and-the-inventoryini-configuration-to-tiup). Then you can upgrade the cluster to v5.0 according to this document.
```

Comment:
Can we use an ordered list for these steps? It looks confusing to me.

Reply:
It seems it's not related to the order: you can switch the order of any item in the list.